test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/InputTagHelperTest.cs

```diff
@@ -1293,6 +1297,217 @@ public void Process_GeneratesFormattedOutput_ForDateTime(string specifiedType, s
         }
 
+        // Html5DateRenderingMode.Rfc3339 is enabled by default.
+        [Theory]
+        [InlineData("Date", "2000-01-02", "date")]
+        [InlineData("DateTime", "2000-01-02T03:04:05.060", "datetime-local")]
+        [InlineData("DateTimeLocal", "2000-01-02T03:04:05.060", "datetime-local")]
+        [InlineData("DateTimeOffset", "2000-01-02T03:04:05.060Z", "text")]
+        [InlineData("Time", "03:04:05.060", "time")]
+        [InlineData("Month", "2000-01", "month")]
+        [InlineData("Week", "1999-W52", "week")]
+        public async Task ProcessAsync_CallsGenerateTextBox_ProducesExpectedValue_ForDateTime(
+            string propertyName,
+            string expectedValue,
+            string expectedType)
+        {
+            // Arrange
+            var expectedAttributes = new TagHelperAttributeList
+            {
+                { "type", expectedType },
+                { "id", propertyName },
+                { "name", propertyName },
+                { "value", expectedValue },
+            };
+
+            var context = new TagHelperContext(
+                tagName: "input",
+                allAttributes: new TagHelperAttributeList(
+                    Enumerable.Empty<TagHelperAttribute>()),
+                items: new Dictionary<object, object>(),
+                uniqueId: "test");
+
+            var output = new TagHelperOutput(
+                expectedType,
+                attributes: new TagHelperAttributeList(),
+                getChildContentAsync: (useCachedResult, encoder) => Task.FromResult<TagHelperContent>(
+                    new DefaultTagHelperContent()))
+            {
+                TagMode = TagMode.SelfClosing,
+            };
+
+            var metadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
+            var model = new DateTime(
+                year: 2000,
+                month: 1,
+                day: 2,
+                hour: 3,
+                minute: 4,
+                second: 5,
+                millisecond: 60,
+                kind: DateTimeKind.Utc);
+
+            var htmlGenerator = HtmlGeneratorUtilities.GetHtmlGenerator(metadataProvider);
+            var tagHelper = GetTagHelper(
+                htmlGenerator,
+                model: model,
+                propertyName: propertyName,
+                metadataProvider: metadataProvider);
+            tagHelper.ViewContext.Html5DateRenderingMode = Html5DateRenderingMode.Rfc3339;
+
+            // Act
+            await tagHelper.ProcessAsync(context, output);
+
+            // Assert
+            Assert.Collection(expectedAttributes,
+                attribute => Assert.Equal(attribute, output.Attributes[0]),
+                attribute => Assert.Equal(attribute, output.Attributes[1]),
+                attribute => Assert.Equal(attribute, output.Attributes[2]),
+                attribute => Assert.Equal(attribute, output.Attributes[3]));
```

Comment:
What goes wrong with `Assert.Equal(expectedAttributes, output.Attributes)` or perhaps `Assert.Equal<TagHelperAttribute>(expectedAttributes, output.Attributes)`.

If you can't do either of those, make `output.Attributes` the first argument and check the elements against `expectedAttributes[0]` and so on.